### ISSUE TITLE
label sampler that do not request specific labels at the center of extracted patches

### ISSUE THREAD

**Issue by MeteorsHub** [fepegar/torchio#560]
**🚀 Feature**
As the title, which means if label label_probabilities is {0(background): 1, 1(foreground): 1}, half the extracted patches will contain foreground (could be anywhere in the patch) and half will only contain background.

**Motivation**
If the foreground can only be at the patch center, the variability of the training patches will be limited. 
Besides, there are cases that the foreground just cannot be at the patch center, like most cases when subject_size is (512, 512, 30) and patch_size is (512, 512, 1). 

Or is the current framework can do this already?

**Comment by fepegar** [comment]
Hi, @MeteorsHub. This is currently not possible with a `LabelSampler`. The patches' centers are used for simplicity.

For that specific case, you could create a custom probability map and use the more generic `WeightedSampler`. For example, say your foreground is in slices 20 to 25 of your 3D tensor. You could do:

```python
probabilities = torch.zeros_like(tensor)
num_voxels_in = 512 * 512 * 5
num_voxels_out = tensor.numel() - num_voxels_in
probability_in = probability_out = 0.5
probability_voxel_in = probability_in / num_voxels_in
probability_voxel_out = probability_out / num_voxels_out
probabilities[..., :20] = probability_voxel_out
probabilities[..., 20:25] = probability_voxel_in
probabilities[..., 25:] = probability_voxel_out
```

Does that make sense? I can post a reproducible example, if necessary.

**Comment by MeteorsHub** [comment]
Thank you for the response. I get your idea. 
Furthermore, I think this feature is very common in applications, so it'll be great you can add this feature to the framework.
